test/EFCore.MySql.FunctionalTests/Query/MatchQueryMySqlTest.cs

```diff
@@ -0,0 +1,245 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query;
+using Microsoft.EntityFrameworkCore.TestUtilities;
+using Pomelo.EntityFrameworkCore.MySql.FunctionalTests.TestUtilities;
+using Pomelo.EntityFrameworkCore.MySql.Query.Expressions.Internal;
+using Xunit;
+
+namespace Pomelo.EntityFrameworkCore.MySql.FunctionalTests.Query
+{
+    public class MatchQueryMySqlTest : MatchQueryMySqlTestBase<MatchQueryMySqlTest.MatchQueryMySqlFixture>
+    {
+        public MatchQueryMySqlTest(MatchQueryMySqlFixture fixture)
+            : base(fixture)
+        {
+        }
+
+        [ConditionalFact]
+        public virtual void Match_In_Natural_Language_Mode()
+        {
+            using var context = CreateContext();
+            var count = context.Set<Herb>().Count(herb => EF.Functions.Match(herb.Name, "First", MySqlMatchSearchMode.InNaturalLanguageMode));
```

Comment:
All of these tests show warning on MySqlMatchSearchMode, stating that it is internal API and it can change in the future. Any way to make it better ?

Reply:
You defined `MySqlMatchSearchMode` in the *internal* namespace `Pomelo.EntityFrameworkCore.MySql.Query.Expressions.Internal`:

https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/c89867b6fe9c514b49fe71dcf31ce904a63f847a/src/EFCore.MySql/Query/Expressions/Internal/MySqlMatchExpression.cs#L9-L17

Just define it in the same namespace as the `DbFunctions` extension methods (`Microsoft.EntityFrameworkCore`), since it is used as a parameter there.

You should also either move the `enum` to the `Extensions\MySqlDbFunctionsExtensions.cs` or create its own file under `Extensions\`.

Reply:
Haven't found enums in official dotnet/efcore repository to take example, so i moved it in separate file and changed namespace.